src/components/filter-radio.jsx

```diff
@@ -0,0 +1,71 @@
+import { connect } from 'react-redux';
+
+function handleChange(e, dispatch, action) {
+  return dispatch(action());
+}
+
+function applyActiveClass(children, filter) {
+  return children == filter ? 'active' : ''
+}
+
+let FilterRadio = ({children, action, filter, dispatch}) => (
+    <label className="filter">
+      <input type="radio"
+        checked={action().type == filter ? 'true' : ''}
+        name="filter"
+        onChange={(e) => handleChange(e, dispatch, action)}
+        className="filter__radio"
```

Comment:
Why do you prefer using this `__` notation of class names? I think using dash case with `style-jsx` is easier and sufficient.

Reply:
It is some established convention and its easier to follow: https://en.bem.info/methodology/naming-convention/#alternative-naming-schemes . It is intended more for global styling cause it adds some namespace. But it can be used with some selectors and regexes to DRY your styles even within a component.